File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.cpp
#include<bits/stdc++.h>
class Solution {
public:
    bool check(vector<int>& nums) {
        int n = nums.size();
        int culprits = 0;
        for (int i = 1; i < n; ++i)
        {
            if (nums[i] < nums[i - 1])
            {
                ++culprits;
            }
        }
        if (nums[0] < nums[n - 1])
        {
            ++culprits;
        }
        return culprits <= 1;        
    }
};

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var check = function(nums) {
    const n = nums.length;
    let count = 1;
    if(n == 1) return true;
    for(let i=1; i<2*n; i++){
        if(nums[(i-1)%n] <= nums[i%n]){
            count++;
        }
        else count = 1;
        if(count == n) return true;
    } 
    return false;
};